### PR TITLE
CLDC-2199 Fix LA/rent soft validation bugs

### PR DESCRIPTION
**Pull request by natdeanlewissoftwire**
ticket: https://digital.dclg.gov.uk/jira/browse/CLDC-2199

image
(the double £ display is a known bug being fixed by another ticket)

This issue is just caused by the soft validation pages not being by the la question. Have also added them after the beds question as that would cause a similar issue. Have put them after relevant setup questions too: 
image

### CODE REVIEW

File: app/models/form/lettings/subsections/setup.rb [hunk 1]
```diff
@@ -10,14 +10,22 @@ def pages
     @pages ||= [
       organisation_page,
       stock_owner_page,
+      Form::Lettings::Pages::MinRentValueCheck.new(nil, nil, self, check_answers_card_number: nil),
```
Comment (jamesrose) [inline app/models/form/lettings/subsections/setup.rb]: You could just make the default value nil here, if required.

Comment (natdeanlewissoftwire) [reply app/models/form/lettings/subsections/setup.rb]: True, I think there's a tendency to do this more explicitly but I'm not that sold on it myself, will make this change

File: app/models/form/lettings/subsections/setup.rb [hunk 2]
```diff
@@ -10,14 +10,22 @@ def pages
...
       Form::Lettings::Pages::Location.new(nil, nil, self),
+      Form::Lettings::Pages::MinRentValueCheck.new(nil, nil, self),
+      Form::Lettings::Pages::MaxRentValueCheck.new(nil, nil, self),
       Form::Lettings::Pages::Renewal.new(nil, nil, self),
       Form::Lettings::Pages::TenancyStartDate.new(nil, nil, self),
+      Form::Lettings::Pages::MinRentValueCheck.new(nil, nil, self),
+      Form::Lettings::Pages::MaxRentValueCheck.new(nil, nil, self),
       Form::Lettings::Pages::RentType.new(nil, nil, self),
+      Form::Lettings::Pages::MinRentValueCheck.new(nil, nil, self),
+      Form::Lettings::Pages::MaxRentValueCheck.new(nil, nil, self),
```
Comment (natdeanlewissoftwire) [inline app/models/form/lettings/subsections/setup.rb]: It seems ugly to have to add the value check so many times to the setup section but all are relevant, have made this field optional to be consistent with how this is handled for sales logs

File: spec/models/form_handler_spec.rb
Comment (arfacamble) [inline spec/models/form_handler_spec.rb:1]: 🥇

File: spec/models/form/lettings/subsections/setup_spec.rb
Comment (arfacamble) [inline spec/models/form/lettings/subsections/setup_spec.rb:1]: 🥉

File: app/models/form/lettings/subsections/setup.rb [hunk 3]
```diff
@@ -10,14 +10,22 @@ def pages
     @pages ||= [
       organisation_page,
       stock_owner_page,
+      Form::Lettings::Pages::MinRentValueCheck.new("stock_owner_min_rent_value_check", nil, self),
+      Form::Lettings::Pages::MaxRentValueCheck.new("stock_owner_max_rent_value_check", nil, self),
```
Comment (arfacamble) [inline app/models/form/lettings/subsections/setup.rb:14]: teh rent soft validations can get triggered by the stock owner?

**Review by arfacamble — APPROVED** [review]
pending question

File: app/models/form/lettings/subsections/setup.rb (hunk 3 above)
Comment (natdeanlewissoftwire) [reply app/models/form/lettings/subsections/setup.rb:14]: yep, it owning org provider type affects lettype which affects rent ranges and hence the validation triggering/not

File: spec/models/form/lettings/subsections/setup_spec.rb
Comment (natdeanlewissoftwire) [reply spec/models/form/lettings/subsections/setup_spec.rb:1]: fair

**Pull request merged**